frontend/src/components/TagsBar/SingleTag/SingleTag.js

```diff
@@ -1,9 +1,14 @@
 import React from 'react';
 import './SingleTag.css';
 
-const SingleTag = ({ tag }) => (
-  <div className="single-tag">
-    {tag}
+const SingleTag = ({ tag, deleteSelTag, showDeleteButton }) => (
+  <div>
+    <div className="single-tag">
+      <div className="tag-name">{tag}</div>
+      {showDeleteButton
+        ? <div className="delete-tag" onClick={() => deleteSelTag(tag)}>x</div> : <div className="hide" />
```

Comment:
Since this is `display: none`, you're effectively inserting a completely invisible div (aka nothing) here. You could just put `null` in the else condition, which will actually insert nothing.